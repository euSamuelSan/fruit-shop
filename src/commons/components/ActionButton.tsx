import React from 'react'
import styled from 'styled-components'
import Colors from '../Colors'

export type ActionButtonProps = {
    className?: string
    primary?: boolean
    color?: string
    disabled?: boolean
    icon?: React.ReactElement
    text: string
    onClick: () => void
}
export default function ActionButton({
    className,
    primary = false,
    color = Colors.leafyGreen,
    disabled = false,
    icon,
    text,
    onClick,
}: ActionButtonProps) {
    return (
        <Button
            className={className}
            primary={primary}
            color={color}
            disabled={disabled}
            onClick={onClick}
        >
            {icon}
            <Text>{text}</Text>
        </Button>
    )
}

const Button = styled.button<Partial<ActionButtonProps>>`
    cursor: pointer;
    border: 2px solid;
    border-radius: 25px;
    border-color: ${props =>
        props.disabled ? Colors.lightBlueGrey : props.color};
    color: ${props => (props.primary ? Colors.white : props.color)};
    background-color: ${props => {
        if (props.disabled) return Colors.lightBlueGrey
        if (props.primary) return props.color
        return 'transparent'
    }};

    width: fit-content;
    height: 35px;
    display: flex;
    align-items: center;
    padding: 0 10px;
    gap: 5px;
`
const Text = styled.p`
    font-size: 14px;
    margin: 0;
`
