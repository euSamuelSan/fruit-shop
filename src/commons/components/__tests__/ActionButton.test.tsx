import { render, RenderResult, screen, fireEvent } from '@testing-library/react'

import ActionButton, { ActionButtonProps } from '../ActionButton'

describe('ActionButton', () => {
    const getProps = (): ActionButtonProps => ({
        text: 'ANY_BUTTON_TEXT',
        onClick: jest.fn(),
    })

    const renderActionButton = (props = getProps()): RenderResult => {
        return render(<ActionButton {...props} />)
    }

    it('Deve desenhar o componente corretamente', () => {
        renderActionButton()

        expect(document.body).toMatchSnapshot()
    })

    it('Deve chamar o onClick ao clicar no botÃ£o', () => {
        const props = getProps()

        renderActionButton(props)
        const button = screen.getByRole('button')
        fireEvent.click(button)

        expect(props.onClick).toHaveBeenCalled()
    })
})
