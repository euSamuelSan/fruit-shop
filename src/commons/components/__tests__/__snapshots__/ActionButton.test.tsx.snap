// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton Deve desenhar o componente corretamente 1`] = `
<body>
  <div>
    <button
      class="sc-bczRLJ hEiFb"
      color="#57bd41"
    >
      <p
        class="sc-gsnTZi fWOnJO"
      >
        ANY_BUTTON_TEXT
      </p>
    </button>
  </div>
</body>
`;
